### PR TITLE
fix: nullreference exception when stopping editor

### PR DESCRIPTION
**Pull request by pbosio**

### CODE REVIEW

**Review by sandrade-dcl — APPROVED** [review]

**Review by AjimenezDCL — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs [hunk 1]
```diff
@@ -145,7 +153,11 @@ void IUsersAroundListHUDListView.Dispose()
 
         userElementDictionary.Clear();
         availableElements.Clear();
-        Destroy(gameObject);
+
+        if (!isGameObjectDestroyed)
+        {
+            Utils.SafeDestroy(gameObject);
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs]: This check shouldn't be necessary. 

`Object.Destroy` doesn't fail when a null object is trying to be destroyed. We can stop using `Utils.SafeDestroy`, I think it doesn't add anything in this case.

Comment (pbosio) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs]: Sorry, you are right...I tried using `SafeDestroy` at first and didn't work... I added the flag and forgot that there 😅

**Review by pravusjif — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs (hunk 1 above)
Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs]: I just saw the changes, we shouldn't remove the `isGameObjectDestroyed` flag as well?

Comment (pbosio) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs]: Nope, It seems that there is a strange condition where the MonoBehaviour is destroyed and not destroyed at the same time? 📦🐈 
So trying to access `gameObject` in this situation arise the exception. I don't know what the `gameObject` property is doing internally 🤷‍♂️
Anyway, that flag is the only thing preventing the null reference exception

Comment (AjimenezDCL) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs]: It's, actually, the other way around.

Unity's shutdown is destroying the `gameobject` holding `UsersAroundListHUDListView` and `UsersAroundListHUDListViewController.Dispose` is doing the same but later. The problem is there's no way for the controller to know that the view was destroyed so, indeed, the flag is needed.

Comment (BrianAmadori) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/UsersAroundListHUD/UsersAroundListHUDListView.cs]: Ugh, this `Destroy` thing on stop is really a nasty issue. Anyway, resolving the discussion.

**Review by BrianAmadori — APPROVED** [review]

**Pull request merged**